tests/engine/training/test_hooks.py

```diff
@@ -0,0 +1,55 @@
+from rasa.engine.caching import Cacheable, TrainingCache
+from rasa.engine.graph import ExecutionContext, GraphNode
+from rasa.engine.model_storage import ModelStorage
+from rasa.engine.training.hooks import TrainingHook
+import rasa.shared.utils.io
+from tests.engine.graph_components_test_classes import (
+    CacheableComponent,
+    CacheableText,
+)
+
+
+def test_training_hook_saves_to_cache(
```

Comment:
we validate that everything is fingerprintable beforehand right? Do we need to handle the case that something is not fingerprintable although the type annotations suggested that something is fingerprintable?

Reply:
I.e. an error within the `.fingerprint` method?

Reply:
yeah, maybe in the `deep_container` one or in the extracted one which we just spoke about